src/coreclr/pal/src/thread/process.cpp

```diff
@@ -1271,7 +1271,7 @@ RaiseFailFastException(
     ENTRY("RaiseFailFastException");
 
     TerminateCurrentProcessNoExit(TRUE);
-    PROCAbort();
+    for (;;) PROCAbort();
```

Comment:
why is continuosly calling required here?

Reply:
Also, aren't such infinite loops `UB` in C++ and some compilers might remove them?

Reply:
> I do not think infinite loops that do something are UB.

AFAIR loops must have a valid exit to not be UB in C++.

Reply:
Link? It is not what "Progress guarantee" in https://en.cppreference.com/w/cpp/language/memory_model says.

Reply:
> Link?

Looking at the table in https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2023/p2809r1.html#effect it seems to be this:
![image](https://github.com/dotnet/runtime/assets/35800402/994241bb-b0ab-4bbf-832a-902621c5bb8c)

I'm not 100% sure how does the compiler know something's side effecting here though.

Reply:
> how does the compiler know something's side effecting here though.

It would have to see all code in `external` and prove that it is side-effect free: no syscall, no writes to volatile variables, etc.  `PROCAbort` is very far from side-effect free.